<center><font color=#555555><font size=6> MAF1731 - Microstructure and Trading Systems </font> <br> <br>
    
<center><font color=#555555><font size=5> Ingeniería Financiera </font> <br> <br> <br> <br>

<center><font color=#555555><font size=6> I.F. Cuauhtémoc Corrales Camacho </font> <br> <br>
<center><font color=#555555><font size=6> I.F. Esteban Marquez Delgado </font> <br> <br>
<center><font color=#555555><font size=6> I.F. Marco Antonio Ochoa Cárdenas </font> <br> <br>
   
<center> <font color= #555555> <font size = 4> Octubre 2022 </a> | <font color= #555555> <font size = 4> Repository: <a href='https://github.com/CuauhtemocCC/MyST_LAB_3_E3'>Link</a></font>

<hr style="border:0.02in solid gray"> </hr>

<center> <font color= #555555> <font size = 6> Lab 3 </font> <br> <br> <font color= #555555> <font size = 5> Comportamiento Financiero  </font>

<hr style="border:0.02in solid gray"> </hr>

# <font color= #6B6B6B> <font size = 6> 1. Introducción </font>

<hr style="border:0.02in solid gray"> </hr>

Dia a día en el mundo de las inversiones, existen mil y un ofertas para colocar nuestros recursos económicos, sin embargo, como podemos estar seguros de que la persona a cargo es un guru de las finanzas o tal vez solamente un tipo con un curso de trading por internet, tampoco es debamos desconfiar de cualquier persona o institución, hay empresas e individuos que han dedicado una enorme cantidad de tiempo a formarse una reputación, pero al final, al final de todo, los que llevan a cabo la toma de decisiones en este mundo son humanos y como humanos inherentemente tienen sentimientos y prejucios parecidos a los de los demás, tampoco es que cualquier persona pueda dedicarse al trading claro está, pero si podemos evaluar y traer a la consciencia los sesgos de conducata más comunes que tienden a cometer los traders, para posteriormente generar estrategias que ayuden a contrarestarlos, pero al mismo tiempo por parte de los clientes poder evalaur a los gestores.


# <font color= #6B6B6B> 2. Objetivos </font>

<hr style="border:0.02in solid gray"> </hr>

Analizar datos sobre posiciones de trading con el fin de detectar sesgos de comportamientos en dichas operaciones.

# <font color= #6B6B6B> 3. Descripción de Datos </font>

<hr style="border:0.02in solid gray"> </hr>

In [2]:
# Importacion de Paqueterias

import data as dt
import main as mn
import visualizations as vs
import pandas as pd

## <font color= #6B6B6B> 3.1 Limpieza de datos: Inversion Pasiva </font>

Para dar inicio con este proyecto se requerió de de una serie de archivos proporcionados por el profesor acerca del comportamiento que el indice NAFTRAC, el cual es un indice creado por BlackRock Ishares que replica el comportamiento de indice más popular de la bolsa de valores mexicana, el IPC.


In [2]:
# Segmento base de datos de NAFTRAC 2020-01-31
dt.data.head(3)

Ticker                      Nombre Clase de activo  Peso (%)  Precio  \
0      AMXL             AMERICA MOVIL L          Equity     13.70   15.69   
1  FEMSAUBD  FOMENTO ECONOMICO MEXICANO          Equity     11.85  170.37   
2  GFNORTEO         GPO FINANCE BANORTE          Equity     10.64  116.09   

         Acciones  Valor de mercado     Valor nominal  \
0  634,706,362.00  9,958,542,819.78  9,958,542,819.78   
1   50,554,056.00  8,612,894,520.72  8,612,894,520.72   
2   66,664,611.00  7,739,094,690.99  7,739,094,690.99   

                         Sector Unnamed: 9          ISIN  \
0                  Comunicación    2667470  MXP001691213   
1  Productos básicos de consumo    2242059  MXP320321310   
2                   Financieros    2421041  MXP370711014   

            Bolsa de valores Ubicación Moneda Mercado de divisas  \
0  Bolsa Mexicana De Valores    México    MXN                MXN   
1  Bolsa Mexicana De Valores    México    MXN                MXN   
2  Bolsa Mexicana De Valores    México    MXN                MXN   

   Tipo de Cambio  
0             1.0  
1             1.0  
2             1.0

El primer paso fue establecer una lista de tickers predeterminada a partir de las empresas que han pertenecido desde el enero de 2020 a julio de 2022 al indice objetivo. Tomando el primer archivos de datos descargados, con la fecha de 31 de enero de 2020, y excluyendo los tickers establecidos en las indicaciones de proyecto, quedo un total de 33 tickers, en los cuales se basaron las futuras extracciones de datos. Ademas de que se realizo una mutacion a dicho listado de tickers mediante un funcion definida como *clean tickers*, con el proposito de que el nombre y estructura de los tickers fuera compatible con la estructura de los mismos en la plataforma de Yahoo Finance.

Como un dato extra, los tickers excluidos no fueron eliminados del portafolio, sino que fueron apartados y fusionados en la variable Cash, la cual para este ejemplo tiene una ponderacion del 3.1% y sera importante para las estrategias de inversion. 

In [3]:
# Segmento lista de Tickers
dt.Tickers.head(3)

0        AMXL.MX
1    FEMSAUBD.MX
2    GFNORTEO.MX
Name: Ticker, dtype: object

El siguiente paso fue extraer los pesos iniciales que cada ticker dispondria en la preparacion del portafolio inicial, estos valores se encontraban en el mismo documento del cual fueron extraidos y mutados los nombres de los activos, se creo un dataframe para juntar tan nombres como poderaciones iniciales, ademas que se ordenaron por orden alfabetico para facilitar la visualizacion de los datos y la futura adesion de los precios descargados.

In [4]:
# Segmento tabla de nombres y pesos
dt.pesos.iloc[0:3,0:2]

Ticker  Peso (%)
14      AC.MX    0.0177
20   ALFAA.MX    0.0151
35  ALPEKA.MX    0.0025

In [5]:
# Grafica de pesos inicial pasiva
vs.bar_chart(dt.pesos,"Ticker","Peso (%)")

Tras esto, se dispuso a realizar la extracción de los datos correspondientes a los nombres en la tabla anterior a traves de la paqueteria yfinance, la cual extrae datos desde el sitio web Yahoo Finance, dicha extracción se hizo para el periodo de enero de 2021 a julio de 2022, con el objetivo de que los periodos equiparables entre ambas estrategias de inversion fuera el mismo. De dicha base de datos extraida, se recorto a solo la columna de precios ajustados y asi la matriz resultante fue de los precios por tickers y en el indice las fechas de esos valores.

Por facilidad ante el siguiente proceso, se decidio por transponer dicha matriz, de tal manera que los nombres de tickers fueran el indice y las fechas sus columnas. Ademas de que se filtro un vector de fechas de cierre, la cual fue extraida a partir del nombre de los archivos compartidos en la carpeta files, con la finalidad de reducir el tamaño de la base de datos extraida y enfocarse en el cierre de mes del periodo de descarga.

In [6]:
# Segmento precios estrategia de inversion pasiva
dt.df2.head(3)

Date       2021-01-29  2021-02-26  2021-03-31  2021-04-30  2021-05-31  \
AC.MX       86.565262   88.146446   93.801521  103.572952  107.946579   
ALFAA.MX    12.846219   11.342515   11.750805   14.134883   13.945353   
ALPEKA.MX   17.533739   18.844303   18.990562   22.466026   22.625360   

Date       2021-06-30  2021-07-30  2021-08-31  2021-09-30  2021-10-26  \
AC.MX      110.410332  115.032234  123.540756  121.680313  120.984344   
ALFAA.MX    14.922925   15.122429   14.065057   14.274536   15.361834   
ALPEKA.MX   24.407906   24.607071   23.581362   23.392153   23.103361   

Date       2021-11-30  2021-12-31  2022-01-26  2022-02-28  2022-03-31  \
AC.MX      126.407181  127.233490  120.497482  131.717651  131.844376   
ALFAA.MX    15.292007   14.982777   14.543866   15.301983   15.110000   
ALPEKA.MX   22.133537   22.213299   24.047791   26.261143   27.090000   

Date       2022-04-29  2022-05-31  2022-06-30  2022-07-29  
AC.MX      129.490005  133.789993  132.500000  141.210007  
ALFAA.MX    13.670000   15.120000   14.280000   14.010000  
ALPEKA.MX   27.549999   26.120001   27.790001   27.110001

## <font color= #6B6B6B> 3.2 Limpieza de datos: Portafolio Eficiente </font>

Para la busqueda del portafolio eficiente se requirio de una nueva extracion de datos desde Yahoo Finance, solo que ahora el periodo de tiempo designado seria de enero de 2020 a enero de 2021, con el fin de establecer un comportamiento previo al periodo de tiempo objetivo determinado para ambas estrategias.

In [7]:
# Segmento datos para la busqueda de portafolio eficiente
dt.prices3.head(3)

AC.MX   ALFAA.MX  ALPEKA.MX   ALSEA.MX    AMXL.MX    ASURB.MX  \
Date                                                                            
2020-01-02  90.031296  16.061821  20.754181  51.480000  14.012444  337.211273   
2020-01-03  90.835075  16.260605  20.902636  51.889999  14.140412  338.867004   
2020-01-06  90.495018  16.434540  20.947174  51.895000  13.975881  332.679016   

            BBAJIOO.MX  BIMBOA.MX  BOLSAA.MX  CEMEXCPO.MX  ...    LABB.MX  \
Date                                                       ...              
2020-01-02   25.797777  33.614262  37.088295         7.16  ...  16.956949   
2020-01-03   26.022106  33.623768  36.309128         7.19  ...  16.965981   
2020-01-06   25.901932  33.771114  36.135986         7.20  ...  16.993065   

            LIVEPOLC-1.MX  MEGACPO.MX     OMAB.MX   ORBIA.MX  PE&OLES.MX  \
Date                                                                       
2020-01-02      89.959549   69.275215  121.228447  37.647270  206.533569   
2020-01-03      91.554382   69.230507  120.657494  38.220154  210.751602   
2020-01-06      92.147797   69.346756  119.473000  38.183784  212.450806   

             PINFRA.MX     RA.MX  TLEVISACPO.MX  WALMEX.MX  
Date                                                        
2020-01-02  181.368134  94.85614      44.418938  53.445782  
2020-01-03  184.611786  94.16571      44.517365  53.028839  
2020-01-06  182.570663  94.36853      43.764416  53.596073  

[3 rows x 33 columns]

## <font color= #6B6B6B> 3.1 Limpieza de datos: Inversion Activa </font>

La preparacion de los datos para la estrategia de inversion pasiva fue similar a los dos aparatdos anteriores, ya que tambien se realizo una extraccion de datos desde Yahoo Finance, con el mismo periodo de tiempo que la inversion pasiva (enero de 2021 a julio 2022), la unica diferencia fue que en vez de recortar por la columna de precios ajustados, se decidio hacerse por la columna de precios de cierre. Este cambio bajo el argumneto de la necesidad de obtener rendimientos mas cercanos a la realidad para la implementacion de esta estrategia. 

Ademas de esto, tambien se filtraron los datos mediante un nuevo vector de fechas para la estrategia y se transpuso la tabla para su futuro analisis e implementacion. 

In [8]:
# Segmento precios estrategia de inversion activa
dt.prices_matrixT.head(3)

Date       2021-01-29  2021-02-26  2021-03-31  2021-04-30  2021-05-31  \
AC.MX           93.07   94.769997  100.849998  108.459999  113.040001   
ALFAA.MX        12.90   11.390000   11.800000   14.170000   13.980000   
ALPEKA.MX       17.66   18.980000   19.070000   22.559999   22.719999   

Date       2021-06-30  2021-07-30  2021-08-31  2021-09-30  2021-10-26  \
AC.MX      115.620003  120.459999  129.369995  125.879997  125.160004   
ALFAA.MX    14.960000   15.160000   14.100000   14.310000   15.400000   
ALPEKA.MX   24.510000   24.709999   23.680000   23.490000   23.200001   

Date       2021-11-30  2021-12-31  2022-01-26  2022-02-28  2022-03-31  \
AC.MX      130.770004  130.520004  123.610001  135.119995      135.25   
ALFAA.MX    15.330000   15.020000   14.580000   15.340000       15.11   
ALPEKA.MX   22.200001   22.280001   24.120001   26.340000       27.09   

Date       2022-04-29  2022-05-31  2022-06-30  2022-07-29  
AC.MX      129.490005  133.789993  132.500000  141.210007  
ALFAA.MX    13.670000   15.120000   14.280000   14.010000  
ALPEKA.MX   27.549999   26.120001   27.790001   27.110001

# <font color= #6B6B6B> 4. Procesos  </font>

<hr style="border:0.02in solid gray"> </hr>

A continuación se describiran los procesos realizados para culminar, tanto las tablas de resultados de las dupla de estrategias de inversion, asi como la generacion del portafolio eficiente que alimenta la estratetgia de inversion activa. Las bases de datos utlizadas para cada proceso ya fueron previamente descritas en el apartado de Descripcion de Datos, asi como las tranformaciones que dichas bases de datos experimentaron para que la implementacion de su onformacion en los procesos fuera mas sencilla y efectiva.

## <font color= #6B6B6B> 4.1 Inversion Pasiva  </font>

<hr style="border:0.02in solid gray"> </hr>

Con la base de datos de precios y fechas transpuesta para esta estrategia, el primer paso a seguir fue la construcion de un nuevo dataframe donde se pudieron registrar los diferentes calculos a ser desarrollados por la estrategia. Dicho dataframe conto en un inicio con tres columnas, "Symbols" que hace referencia a los nombres de los activos en el portafolio, "Pesos" que hace referencia a los precios inciales inluidos en el NAFTRAC y "Precio" el cual es el precio de los activos extraidos desde Yahoo Finance en el mes de inicio (enero 2021).

Despues, se agregaron dos columnas fundamentales, "Postura" la cual es la multiplicacion de los pesos por el capital inicial (1,000,000) y "Titulos", el cual hace alusion a la cantidad de titulos por activo presente en el portafolio. Su calculo para este caso fue la division de la "Postura" sobre el "Precio" del periodo. 

In [9]:
# Segmento tabla de calculo de inversion pasiva
dt.Base.head(3)

Symbols   Pesos  Precio   Postura  Titulos
AC.MX          AC.MX  0.0177  141.21  27959.58    198.0
ALFAA.MX    ALFAA.MX  0.0151   14.01  15957.39   1139.0
ALPEKA.MX  ALPEKA.MX  0.0025   27.11   3741.18    138.0

El siguiento paso fue como tal la creacion de un ciclo iterativo donde para cada mes siguiente en la matriz de precios correspondiente se remplazaria la columna de "Precio" en la base de calculo y se recalcularia la columna de "Postura", multiplicando los valores nuevos de "Precio" por la cantidad fija dentro de la columna "Titulos". 

Los resultados de dicha iteracion se extran como diccionario indexado cuyas variables **Key** son las fechas dentro del vector de fechas y el capital nuevo, el cual se calculo como la suma de toda la columna de posturas. Ademas, como dato extra, se transfomo el diccionario a un nuevo dataframe y se añadieron dos columnas nuevas que hacen referencia al rendimiento mensual y al rendimiento acumulado mensual dentro del periodo. 

In [10]:
# Segmento tabla de invesion pasiva
dt.inv_pas_df.head(3)

Timestamp    Capital   Yield  Acum Yield
0  2021-01-29   969000.0  0.0000      0.0000
1  2021-02-26   969355.0  0.0366      0.0366
2  2021-03-31  1026917.0  5.9382      5.9748

## <font color= #6B6B6B> 4.1 Busqueda de Portafolio Eficiente  </font>

Este proceso fue distinto al seguido por la estrategias pero queda compensado por su brevedad. Con la data preparada para esta busqueda, se calcularon los rendimientos logaritmicos que cada precio en dicha base. Esto con la finalidad de obtener un vector de rendimientos promedio por activo y una matriz de covarianza entre los activos, pque seran utilizadas para el calculo de los pesos eficientes. 

El calculo de los pesos eficientes se requeria a traves de la maximizacion del sharpe estandar, y esto se realizo mediante la redaccion de funciones implicitas (factor **lambda**) y explicitas (**definicion**). La funcion definida para el calculo de los precios eficientes es *Portfolio* y se alimenta de tres argumentos, una vector de rendimientos, una matriz de covarianza y una tasa de referencia, que para este ejercicio, fue del 4.29%.

El resultado de esta funcion arrojo un vector de pesos eficiente que coincide con el orden alfabetico de los tickers. 

In [11]:
# Vector de pesos eficientes
dt.vector_pesos

array([ 0.  ,  0.  ,  0.  ,  0.  ,  0.  ,  0.  ,  0.  ,  0.  ,  0.  ,
        1.95, 40.6 ,  0.  ,  0.  ,  0.  ,  0.  ,  0.  ,  0.  ,  0.  ,
        0.  , 40.53,  2.03,  0.  ,  0.  ,  0.  ,  0.  ,  0.  ,  0.  ,
        0.  , 14.89,  0.  ,  0.  ,  0.  ,  0.  ])

## <font color= #6B6B6B> 4.1 Inversion Activa  </font>

El proceso de esta estrategia fue el mas complejo del proyecto, ya que requeria de algunas condiciones que subieron de dificultad su creacion. Al igual que para la estrategia pasiva, el primer paso fue la creacion de una base de calculos que en un inicio contenian las mismas columnas de salida que la base en la estrategia pasiva ("Symbols","Pesos","Precio","Postura","Titulos"), con las diferencia de que la columna de pesos ahora fue reemplazada por el vector de pesos eficientes obtenidos y la informacion de precios se extrae de una matriz de precios distinta. 

Ademas de esas cinco columnas se crearon nuevas bases con matrices de ceros que serian de apoyo para la realizacion de los calculos, estas fueron "Precios Nuevos", "Titulos Nuevos", "Diferencial de Titulos", "Diferencial Cash", "Comision", "Pesos Nuevos" y "Titulos de Compra". 

Las condiciones de la inversion pasiva son las siguientes:
+ Aumentar la cantidad de titulos de un activo 2.5% si el rendimiento de un mes contra otro es de equivalente o superior al 5%. 
+ Reducir la cantidad de titulos de un activo 2.5% si el rendimiento de un mes contra otro es de equivalente o inferior al -5%.
+ Cada compra de titulos requiere del pago de una comision igual al 0.00125 o 0.125% del valor de la operacion. 

Para lograr este proposito se creo un metodo interativo indexado, donde se tomaran en cuenta las columnas de la matriz de precios y la filas de la base de calculos.

Las columnas de ceros fueron llenadas de la siguiente manera:
+ "Precios Nuevos": Extraer datos de los precios del mes siguiente para compararlos con los precios en el mes previo.
+ "Titulos Nuevos": Segun los redimientos mensuales de cada activo, se recalculan la cantidad de titulos siguiendo las condiciones de la estrategia.
+ "Diferencial de Titulos": Es la resta entre la cantidad de titulos nueva y la cantidad previa.
+ "Diferencial Cash" Es la expresion de los titulos nuevos pero en su valor en cash. Las operacion de compra restan y las de venta suman. 
+ "Comision": Es el pago de comision a hacerse en la operaciones de compra. 
+ "Pesos Nuevos": Es el recalculo de los precios en base a la nueva postura que traen consigo los titulos nuevos.
+ "Titulos de Compra" Es el conteo de titulos de compra para cada fecha.

De esta manera los datos en la tabla se actualizan para cada mes en el vector de fechas y se guardaron dichos resultados en un nuevo diccionario.


In [12]:
# Segmento tabla de invesion activa
dt.inv_act_df.head(3)

Timestamp    Capital   Yield  Acum Yield
0  2021-01-29   969000.0  0.0000      0.0000
1  2021-02-26  1017718.0  5.0277      5.0277
2  2021-03-31  1099097.0  7.9962     13.0239

## <font color= #6B6B6B> 4.1 Tabla de Operaciones  </font>

Los numeros obtenidos para esta tabla de salida se encuentran dentro del proceso de estrategia activa, ya que provienen del conteo de titulos nuevos comprados y pagos por comision experimentados en la base de calculos de la estrategia de inversion activa. Estos datos tambien fueron almacenados en un diccionario con las **Key** "Titulos de Compra" y "Comision". Luego fueron convertidos a un dataframe para una lectura mas sencilla. Y en esta dataframe se calcularon dos nuevas columnas, "Titulos Totales" el cual es la suma acumulado de los titulos a lo largo de los meses y "Comision Acumulada" siendo esta la suma acumulada de los pagos por comisiones.

In [13]:
# Segmento Tabla de Operaciones
dt.operaciones_df.head(3)

Timestamp  Titulos Totales  Titulos Compra  Comision  Comision Acum
0  2021-01-29              0.0             0.0     0.000          0.000
1  2021-02-26            152.0           152.0    12.885         12.885
2  2021-03-31            494.0           342.0    26.985         39.870

## <font color= #6B6B6B> 4.1 Tabla de Desempeño  </font>

La tabla de desempeño fue creada como un dataframe que almacena y expone variables creadas que califican el comportamiento de cada inversion. Dichas variables son el rendimiento mensual promedio, el rendimiento mensual acumulado final y el indice de sharpe; cada variable fue aplicada a los resultados de cada estrategia de inversion.

In [14]:
# Muestra Tabla de Desempeño
dt.tabla_des

Medida             Descripcion  Inv Pasiva  Inv Activa
0  rend_m   Rend Mensual Promedio    0.959984   -0.080147
1  rend_c  Rend Acumulado Mensual   18.239700   -1.522800
2  sharpe        Indice de Sharpe   -0.710300   -0.560835

# <font color= #6B6B6B> <font size = 6> 5. Conclusiones </font>

<hr style="border:0.02in solid gray"> </hr>

Para dar apropiado incio a este apartado se presentaran las tablas de resultados obtenidas tras los distintos procesos empleados. El primer punto a destacar es la comparacion de los comportamientos simulados y los rendimientos obtenidos por parte de cada estratetegia. 

El proposito de que tanto al estrategia activa como la pasiva compartieran un mismo periodo de tiempo es que la comparacion de rendimientos fuera equiparable. Y al comparar tenemos los siguientes puntos clave:

+ Ambas estrategias inician con un captital de 1,000,000 menos el valor del cash, lo que deja una cantidad de 969,000, y aunque en un inicio los rendimientos obtenidos por parte de la estrategia activa fueron mayores, al termino del periodo de simulacion, es la estrategia pasiva la cual posee un mayor rendimiento acumulado final, lo cual quiere decir que a lo largo de los meses, la estrategia pasiva tuvo rendimientos positivos en mayor cantidad que la estrategia activa. Este hecho se ve impactado en el capital, ya que mientras la estrategia pasiva tiene un valor de capital de 1,139,191, lo cual siginifica una ganancia de 139,191 o un rendimiento a plazo de 14.34%; el capital de la estrategia activa a termino es de 903,171, con lo que se experimento una perdida de 65,829, es decir una perdida del 6.79% del capital inicial. Con esto queda mas que claro que la estrategia pasiva fue mejor.

+ Un aspecto a considerar aqui fue la diversificacion del portafolio, ya que mientras el portafolio de la estrategia pasiva contenia 33 activos, el portafolio de la inversion activa solo contiene 5 activos, y aunque estos 5 activos son aquellos que generan mayores rendimientos, no quiere decir que los rendimientos sean constantes. Aqui se confirma el principio de la diversificacion, donde un portafolio mas distribuido compensa el riesgo de perdidas. 

In [15]:
# Tabla de Estrategias, izquierda Pasiva, derecho Activa, MESES INICIALES
pd.concat([dt.inv_pas_df,dt.inv_act_df],axis=1).head()

Timestamp    Capital   Yield  Acum Yield   Timestamp    Capital   Yield  \
0  2021-01-29   969000.0  0.0000      0.0000  2021-01-29   969000.0  0.0000   
1  2021-02-26   969355.0  0.0366      0.0366  2021-02-26  1017718.0  5.0277   
2  2021-03-31  1026917.0  5.9382      5.9748  2021-03-31  1099097.0  7.9962   
3  2021-04-30  1063220.0  3.5351      9.5099  2021-04-30  1027883.0 -6.4793   
4  2021-05-31  1136349.0  6.8781     16.3880  2021-05-31  1104815.0  7.4845   

   Acum Yield  
0      0.0000  
1      5.0277  
2     13.0239  
3      6.5446  
4     14.0291

In [16]:
# Tabla de Estrategias, izquierda Pasiva, derecho Activa, MESES FINALES
pd.concat([dt.inv_pas_df,dt.inv_act_df],axis=1).tail()

Timestamp    Capital   Yield  Acum Yield   Timestamp    Capital    Yield  \
14  2022-03-31  1306452.0  6.3145     31.2525  2022-03-31  1157536.0   4.7964   
15  2022-04-29  1196450.0 -8.4199     22.8326  2022-04-29  1032970.0 -10.7613   
16  2022-05-31  1218811.0  1.8689     24.7015  2022-05-31   974680.0  -5.6430   
17  2022-06-30  1128418.0 -7.4165     17.2850  2022-06-30   868277.0 -10.9167   
18  2022-07-29  1139191.0  0.9547     18.2397  2022-07-29   903171.0   4.0188   

    Acum Yield  
14     21.7794  
15     11.0181  
16      5.3751  
17     -5.5416  
18     -1.5228

In [17]:
# Grafica variacion de capital Estrategia Pasiva
vs.desempeño_lineal(dt.inv_pas_df,"Capital")

In [18]:
# Grafica variacion de capital Estrategia Activa
vs.desempeño_lineal(dt.inv_act_df,"Capital")

In [19]:
# Grafica redimientos mensuales Estrategia Pasiva
vs.desempeño_lineal(dt.inv_pas_df,"Yield")

In [20]:
# Grafica redimientos mensuales Estrategia Activa
vs.desempeño_lineal(dt.inv_act_df,"Yield")

+ Por otro lado, se debe recordar la variable de alamcenamiento de dinero llamada cash, la cual para el caso pasivo no cambia su valor de 31,000, pero para el cambia activo si es fluctuante mes a mes debido a las operaciones, dejando un fondo de 52,366.2 de cash para el caso activo. Aun con esto, no fue suficiente el cash para aplacar las perdidas en el caso activo, ya que la suma del capital final mas el cash final seria de 955,537, lo cual solo reduce la perdida del 6.79% del capital inicial al 4.44% del capital inicial. 

+ En cambio, la suma del capital final mas el cash para el caso pasivo es de 1,170,191. 

In [21]:
# Balance final de Cash, estrategia activa
round(dt.cash,2)

52366.2

+ Respecto a la tabla de operaciones, se tiene como un dato interesante que la estrategia activa simulo la compra de 1,779 titulos a lo largo de 18 meses de actividad, lo cual trajo consigo un pago de comisiones de 145.80, este ultimo no es visto como apreciable para el cash, no obstante con este ultimo se puede obtener una conclusion particular, que en el periodo hubo mas operaciones de venta de titulos que de compra. Esta afirmacion se basa en el hecho de que la cuenta de cash paso de tener un valor de 31,000 a inicio del periodo a 52,366 al termino, y como las operaciones de venta suman efectivo a la cuenta de cash, esto quiere decir que entre operaciones de venta y compra hubo un diferencia de 21,366 a favor de las ventas. Y si nos reiteramos a las condiciones de la estrategia activa, esto quiere decir que durante el periodo, los activos presentes tuvieron mas rendimientos menores al -5% que su par positivo.

In [22]:
# Tabla de Operaciones
dt.operaciones_df.tail()

Timestamp  Titulos Totales  Titulos Compra  Comision  Comision Acum
14  2022-03-31           1511.0           117.0   15.3662       130.6512
15  2022-04-29           1511.0             0.0    0.0000       130.6512
16  2022-05-31           1511.0             0.0    0.0000       130.6512
17  2022-06-30           1511.0             0.0    0.0000       130.6512
18  2022-07-29           1779.0           268.0   15.1538       145.8050

Finalmente con la tabla de desempeño, se puede apreciar que los numeros de rendimiento mensual promedio y rendmiento mensual acumulado son mejores para la estrategia de inversión pasicva, sin embargo, esto cambia al hablar del radio de Sharpe. Primeramente, un radio de Sharpe negativo no siginifica que el portafolio no posea cualificaciones de rendimientos, sino que para que periodo de tiempo designado, la tasa libre de riesgo tiene un rendimiento mayor que el portafolio mismo. Esto no significa que alguna de las dos estrategias sea mala, como ya se expuso, la estrategia pasiva dio rendimientos siginificativos, solo que para el periodo de tiempo asignado, no se recomienda el uso de ese portafolio, a mi recomendacion, yo guardaria este tipo de portafolio para otro momento en el cual el mercado de valores mexicano tenga rendimientos mas favorables a los presentados. 

In [23]:
# Tabla de Desempeño
dt.tabla_des

Medida             Descripcion  Inv Pasiva  Inv Activa
0  rend_m   Rend Mensual Promedio    0.959984   -0.080147
1  rend_c  Rend Acumulado Mensual   18.239700   -1.522800
2  sharpe        Indice de Sharpe   -0.710300   -0.560835

Como un hecho extra, me gustaria añadir para la parte de la estartegia activo como las posturas cambiantes afectaron a la distribucion de pesos en el portafolio. Los mayores cambios se observan para las empresas "CUERVO.MX" Y "PE&OLES.MX", donde la primera paso de tener un peso inicial de 40.6% a un peso final de 45.8%, lo cual implica una aumento de 5.2 pp, mientras que para la segunda, paso de un peso incial de 14.9% a un peso final de 9.4, una caida de 5.5 pp. 

In [24]:
# Distribucion Eficiente Inicial
vs.pie_chart(dt.pesos_noceros,"Ticker","Pesos Eficientes")

In [25]:
# Distribucion Eficiente Final
vs.pie_chart(dt.pesos_noceros2,"Symbols","Pesos_Nuevos")

Fin

<br>

# <font color= #6B6B6B> <font size = 6> 6. References </font>

<hr style="border:0.02in solid gray"> </hr>

[1] Munnoz, 2020. Proyecto Laboratorio 1. https://github.com/CuauhtemocCC/Microestructuras-de-Trading-Lab-1. (2022).

<br>